### PR TITLE
CogVideoX benchmarks with fp8dqrow

### PR DESCRIPTION
**Pull request by a-r-r-o-w**

### CODE REVIEW

File: inference/benchmark_video.py [hunk 1]
```diff
@@ -154,7 +156,7 @@ def get_args():
         "--model_id",
         type=str,
         default="THUDM/CogVideoX-5b",
-        choices=["THUDM/CogVideoX-2b", "THUDM/CogVideoX-5b"],
+        # choices=["THUDM/CogVideoX-2b", "THUDM/CogVideoX-5b"],
```
Comment (a-r-r-o-w) [inline inference/benchmark_video.py:159]:
```suggestion
        choices=["THUDM/CogVideoX-2b", "THUDM/CogVideoX-5b"],
```

File: inference/benchmark_video.py [hunk 2]
```diff
@@ -33,7 +34,8 @@
     "bf16": lambda module: module.to(dtype=torch.bfloat16),
     "fp8wo": lambda module: quantize_(module, float8_weight_only()),
     "fp8dq": lambda module: quantize_(module, float8_dynamic_activation_float8_weight()),
-    "fp6": lambda module: quantize_(module, fp6_llm_weight_only()),
+    "fp8dqrow": lambda module: quantize_(module, float8_dynamic_activation_float8_weight(granularity=PerRow())),
+    "fp6": lambda module: quantize_(module, fpx_weight_only(3, 2)),
```
Comment (sayakpaul) [inline inference/benchmark_video.py:38]: Let's do this in a separate PR for both `benchmark_video.py` and `benchmark_image.py`?

File: README.md [hunk 3]
```diff
@@ -277,6 +277,22 @@ The table below provides some benchmarks:
 
 Through visual inspection of various outputs, we identified that the best results were achieved with int8 weight-only quantization, int8 dynamic quantization, fp8 (currently supported only on Hopper architecture), and autoquant. While the outputs sometimes differed visually from their standard fp16/bf16 counterparts, they maintained the expected quality. Additionally, we observed that int4 dynamic quantization generally produced satisfactory results in most cases, but showed greater deviation in structure, color, composition and motion.
 
+With the newly added `fp8dqrow` scheme, the inference latency is **76.70 seconds** for CogVideoX-5b (batch size: 1 , steps: 50, frames, resolution: 720x480) on an H100. `fp8dqrow` has more scales per tensors and less quantization error. The quality, from visual inspection, is very close to fp16/bf16 and better than int8 in many cases.
```
Comment (a-r-r-o-w) [inline README.md:280]:
```suggestion
With the newly added `fp8dqrow` scheme, the inference latency is **76.70 seconds** for CogVideoX-5b (batch size: 1 , steps: 50, frames: 49, resolution: 720x480) on an H100. `fp8dqrow` has more scales per tensors and less quantization error. The quality, from visual inspection, is very close to fp16/bf16 and better than int8 in many cases.
```

**Comment by a-r-r-o-w** [comment]
>  Also, did you check the output? With FP8, I would expect better quality preservation.

Made a note of that in the README already :) Quality is very good!

> The quality, from visual inspection, is very close to fp16/bf16 and better than int8 in many cases.

File: inference/benchmark_video.py [hunk 4]
```diff
@@ -17,9 +17,10 @@
     int4_weight_only,
     float8_dynamic_activation_float8_weight,
     float8_weight_only,
+    fpx_weight_only,
 )
+from torchao.quantization.quant_api import PerRow
 from torchao.sparsity import sparsify_
```
Comment (a-r-r-o-w) [inline inference/benchmark_video.py:23]:
```suggestion
)
from torchao.prototype.quant_llm import fp6_llm_weight_only
from torchao.quantization.quant_api import PerRow
from torchao.sparsity import sparsify_
```

**Comment by sayakpaul** [comment]
Feel free to also edit the percentage here:

> We provide end-to-end inference and experimental training recipes to use torchao with diffusers in this repo. We demonstrate 53.88% speedup on [Flux.1-Dev](https://huggingface.co/black-forest-labs/FLUX.1-dev)* and 21% speedup on [CogVideoX-5b](https://huggingface.co/THUDM/CogVideoX-5b) when comparing compiled quantized models against their standard bf16 counterparts**.

And then merge!

File: inference/benchmark_video.py (hunk 2 above)
Comment (a-r-r-o-w) [inline inference/benchmark_video.py:38]:
```suggestion
    "fp6": lambda module: quantize_(module, fp6_llm_weight_only()),
```

**Comment by a-r-r-o-w** [comment]
> Feel free to also edit the percentage here:

Actually, fp8dqrow is a few milliseconds slower than plain fp8 here. So, I think that's still the fastest and the numbers should be the same

**Comment by sayakpaul** [comment]
Ah okay then. All good?
